core/drivers/bcm_sotp.c

```diff
@@ -125,6 +143,137 @@ TEE_Result bcm_iproc_sotp_mem_read(uint32_t row_addr, uint32_t sotp_add_ecc,
 	return ret;
 }
 
+TEE_Result bcm_iproc_sotp_mem_write(uint32_t row_addr, uint32_t sotp_add_ecc,
```

Comment:
It looks like `sotp_add_ecc` is used as a `bool`. Given how it's compared with a `1` it seems a bit fragile. How about making it a `bool` instead?